cloudformation/stacks/AutoSpotting/template.yaml

```diff
@@ -0,0 +1,262 @@
+              Ref: "OnDemandPriceMultiplier"
+            REGIONS:
+              Ref: "Regions"
+            SPOT_PRICE_BUFFER_PERCENTAGE:
+              Ref: "SpotPricePercentageBuffer"
+            SPOT_PRODUCT_DESCRIPTION:
+              Ref: "SpotProductDescription"
+            TAG_FILTERING_MODE:
+              Ref: "TagFilteringMode"
+            TAG_FILTERS:
+              Ref: "FilterByTags"
+        Handler:
+          Ref: "LambdaHandlerFunction"
+        MemorySize:
+          Ref: "LambdaMemorySize"
+        Role:
+          Fn::GetAtt:
+            - "LambdaExecutionRole"
+            - "Arn"
+        Runtime: "go1.x"
+        Timeout: "300"
+      Type: "AWS::Lambda::Function"
+    LambdaPolicy:
+      Properties:
+        PolicyDocument:
+          Statement:
+            -
+              Action:
+                - "autoscaling:DescribeAutoScalingGroups"
+                - "autoscaling:DescribeLaunchConfigurations"
+                - "autoscaling:AttachInstances"
+                - "autoscaling:DetachInstances"
+                - "autoscaling:DescribeTags"
+                - "autoscaling:UpdateAutoScalingGroup"
+                - "ec2:CreateTags"
+                - "ec2:DescribeInstances"
+                - "ec2:DescribeRegions"
+                - "ec2:DescribeSecurityGroups"
+                - "ec2:DescribeSpotInstanceRequests"
+                - "ec2:DescribeSpotPriceHistory"
+                - "ec2:RequestSpotInstances"
+                - "ec2:TerminateInstances"
+                - "iam:PassRole"
+                - "iam:CreateServiceLinkedRole"
+                - "logs:CreateLogGroup"
+                - "logs:CreateLogStream"
+                - "logs:PutLogEvents"
+              Effect: "Allow"
+              Resource: "*"
+        PolicyName: "LambdaPolicy"
+        Roles:
+          -
+            Ref: "LambdaExecutionRole"
+      Type: "AWS::IAM::Policy"
+    LogGroup:
+      Properties:
+        LogGroupName:
+          Fn::Join:
+            - ""
+            -
+              - "/aws/lambda/"
+              -
+                Ref: "LambdaFunction"
+        RetentionInDays:
+          Ref: "LogRetentionPeriod"
+      Type: "AWS::Logs::LogGroup"
+    PermissionForEventsToInvokeLambda:
+      Properties:
+        Action: "lambda:InvokeFunction"
+        FunctionName:
+          Ref: "LambdaFunction"
+        Principal: "events.amazonaws.com"
+        SourceArn:
+          Fn::GetAtt:
+            - "ScheduledRule"
+            - "Arn"
+      Type: "AWS::Lambda::Permission"
+    ScheduledRule:
+      Properties:
+        Description: "ScheduledRule for launching the AutoSpotting Lambda function"
+        ScheduleExpression:
+          Ref: "ExecutionFrequency"
+        State: "ENABLED"
+        Targets:
+          -
+            Arn:
+              Fn::GetAtt:
+                - "LambdaFunction"
+                - "Arn"
+            Id: "AutoSpottingEventGenerator"
+      Type: "AWS::Events::Rule"
+
```

Comment:
As JSON is a subset of YAML, I assume you made an automatic conversion?

Reply:
Yes, I used `json2yaml template.json > template.yaml`